### PR TITLE
HLL params.

### PR DESCRIPTION
**Pull request by claygoddard**
Add support for HLL params for testing various accuracy levels but
also to unlock options for cheap unique estimations for facet sorting.

### CODE REVIEW

**Comment by claygoddard** [comment]
Want to use this to try and address https://app.shortcut.com/fullstory/story/172044/enhance-solr-with-a-cheap-unique-approximation-appropriate-for-prelim-sorting

File: solr/core/src/java/org/apache/solr/search/ValueSourceParser.java [hunk 1]
```diff
@@ -987,7 +987,14 @@ public ValueSource parse(FunctionQParser fp) throws SyntaxError {
     addParser("agg_hll", new ValueSourceParser() {
       @Override
       public ValueSource parse(FunctionQParser fp) throws SyntaxError {
-        return new HLLAgg(fp.parseArg());
+        String field = fp.parseArg();
+        if (fp.hasMoreArguments()) {
+          int log2m = fp.parseInt();
```
Comment (hiteshk25) [inline solr/core/src/java/org/apache/solr/search/ValueSourceParser.java:992]: minor: do we want some validation here(its integer argument?)

**Review by hiteshk25 — APPROVED** [review]
LGTM

File: solr/core/src/java/org/apache/solr/search/ValueSourceParser.java (hunk 1 above)
Comment (claygoddard) [reply solr/core/src/java/org/apache/solr/search/ValueSourceParser.java:992]: The `FunctionQParser` under hood does the validation at parse time.

**Pull request merged**